xla/service/gpu/nccl_clique.h

```diff
@@ -147,7 +147,8 @@ struct NcclClique : public Lockable<NcclCliqueCommunicators, NcclCliqueName> {
 absl::StatusOr<std::shared_ptr<NcclClique::Lock>> AcquireNcclClique(
     RunId run_id, NcclCliqueKey clique_key,
     const NcclCliqueIdCallback& clique_id_callback, int32_t rank,
-    size_t num_local_participants, bool may_skip_rendezvous);
+    size_t num_local_participants, bool may_skip_rendezvous,
+    int64_t collective_max_nchannels, int64_t p2p_max_nchannels);
```

Comment:
pass `std::optional<NcclCommConfig>`? you have to construct it anyway, and nccl_api header dependency is already here